src/arr/compiler/anf-loop-compiler.arr

```diff
@@ -919,15 +919,18 @@ fun compile-split-app(l, compiler, opt-dest, f, args, opt-body, app-info, is-def
               j-expr(j-dot-assign(RUNTIME, "EXN_STACKHEIGHT", j-num(0))),
               j-expr(j-assign(ans, rt-method("makeCont", cl-empty)))]))] +
         CL.map_list2(
-          lam(compiled-arg, arg): j-expr(j-assign(arg, compiled-arg)) end,
+          lam(compiled-arg, arg): j-expr(j-assign(compiler-name(arg.toname()), compiled-arg)) end,
```

Comment:
Should this be `j-var` of this new name? I worry that this will make a bunch of global variables.

Reply:
Also, do we have a guarantee that these names won't conflict with other existing names in scope (or in future modifications to the compiler?) These seem a little brittle if they just use the arguments names with a consistent prefix appended, which is what `compiler-name` does, and `toname` just gives back a human-looking name.

Reply:
The newer version generates a new id using `fresh-id` and now uses `j-var`.